### ISSUE TITLE
What preprocessing required for mix type of dataset ( continuous and categorical ) ?

### ISSUE THREAD

**Issue by Abhinav43** [yzhao062/pyod#366]
Hello Pyod developers: I have one question regarding the outliers detection

If my dataset is the combination of continuous ( numerical ) and categorical values, What are the steps to feed this data to pyod models, and is there any preprocessing required for continuous ( numerical ) and categorical values such as encoding ( categorical columns ), normalization ( continuous)?

**Comment by yzhao062** [comment]
Thanks for checking. PyOD can only handle numerical values (similar to scikit-learn we ask users to handle the data type conversion). As long as your data is fully numerical, PyOD is compatible. For instance, if you use one hot encoding for conversion, leading to high-dimensional datasets, try isolation forests, ecod, copod, or hbos in pyod as the detection algorithm. These are fast with high dimensional datasets.

**Comment by Abhinav43** [comment]
Thanks for the reply, if I have continuous ( numerical ) and categorical values I'll encode categorical data, now : 
1) do I need to specify which column is categorical and which one is continuous or should I feed this encoded data directly to pyod?
2) do I need to normalize the continuous values? if yes then should I normalize only continuous or encoded categorical values too?

**Comment by bhralzz** [comment]
Dear Zhao,
could you please reply above question?

**Comment by yzhao062** [comment]
> Thanks for the reply, if I have continuous ( numerical ) and categorical values I'll encode categorical data, now :
> 
> 1. do I need to specify which column is categorical and which one is continuous or should I feed this encoded data directly to pyod?
> 2. do I need to normalize the continuous values? if yes then should I normalize only continuous or encoded categorical values too?

1. no need to specify that. as long as they are numerical, feed them in directly.
2. column-wise standardization or min-max scalar by sklearn is sufficient. For some models, e.g., Forest, this is not needed but compatible.

**Comment by yzhao062** [comment]
> Dear Zhao, could you please reply above question?

just did.
